File: 5747.py
def solution(X_list, Y_list):
    cnt_xOdd, cnt_yEven = 0, 0
    for i in range(len(X_list)):
        if X_list[i] % 2 != 0:
            cnt_xOdd += 1
    for i in range(len(Y_list)):
        if Y_list[i] % 2 == 0:
            cnt_yEven += 1
    sol = abs(cnt_xOdd - cnt_yEven)
    return sol


if __name__ == '__main__':
    N = int(input())
    sol_list = []
    while N:
        X_list = list(map(int, input().split()))
        Y_list = list(map(int, input().split()))
        sol = solution(X_list, Y_list)
        sol_list.append(str(sol))
        N = int(input())
    print('\n'.join(sol_list))

File: 5747.cpp
#include <iostream>
using namespace std;
#define MAX_N 100

int n;
int arr_X[MAX_N] = { 0 }, arr_Y[MAX_N] = { 0 };
int sol = 0;

void input() {
    cin >> n;
    int num;
    for (int i = 0; i < n; i++) {
        cin >> num;
        arr_X[i] = num;
    }
    for (int i = 0; i < n; i++) {
        cin >> num;
        arr_Y[i] = num;
    }
}
void compute() {
    int cnt_xOdd = 0, cnt_yEven = 0;
    for (int i = 0; i < n; i++)
        if (arr_X[i] % 2 != 0)
            cnt_xOdd++;
    for (int i = 0; i < n; i++)
        if (arr_Y[i] % 2 == 0)
            cnt_yEven++;
    sol = abs(cnt_xOdd - cnt_yEven);
}
void output() {
    cout << sol << '\n';
}
int main() {
    while (1) {
        input();
        if (n == 0)
            break;
        compute();
        output();
    }
    return 0;
}